kernel/src/deferred_call.rs

```diff
@@ -117,19 +134,43 @@ impl DynDefCallRef<'_> {
 // thread. TODO: Once Tock decides on an approach to replace `static mut` with
 // some sort of `SyncCell`, migrate all three of these to that approach
 // (https://github.com/tock/tock/issues/1545).
-static mut CTR: Cell<usize> = Cell::new(0);
+static CTR: SingleThreadValue<Cell<usize>> = SingleThreadValue::new(Cell::new(0));
 
 /// This bitmask tracks which of the up to 32 existing deferred calls have been
 /// scheduled. Any bit that is set in that mask indicates the deferred call with
 /// its [`DeferredCall::idx`] field set to the index of that bit has been
 /// scheduled and not yet serviced.
-static mut BITMASK: Cell<u32> = Cell::new(0);
+static BITMASK: SingleThreadValue<Cell<u32>> = SingleThreadValue::new(Cell::new(0));
 
 /// An array that stores references to up to 32 `DeferredCall`s via the low-cost
 /// [`DynDefCallRef`].
 // This is a 256 byte array, but at least resides in `.bss`.
-static mut DEFCALLS: [OptionalCell<DynDefCallRef<'static>>; 32] =
-    [const { OptionalCell::empty() }; 32];
+static DEFCALLS: SingleThreadValue<MapCell<[OptionalCell<DynDefCallRef<'static>>; 32]>> =
+    SingleThreadValue::new(MapCell::new([const { OptionalCell::empty() }; 32]));
+
+/// Initialize the static state used by deferred calls.
+///
+/// This ensures it can safely be used as a global variable.
+#[cfg(target_has_atomic = "ptr")]
+pub fn initialize_deferred_call_state<P: ThreadIdProvider>() {
+    CTR.bind_to_thread::<P>();
+    BITMASK.bind_to_thread::<P>();
+    DEFCALLS.bind_to_thread::<P>();
+}
+
+/// Initialize the static state used by deferred calls.
+///
+/// This ensures it can safely be used as a global variable.
+///
+/// # Safety
+///
+/// Callers of this function must ensure that this function is never called
+/// concurrently with other calls to [`initialize_deferred_call_state_unsafe`].
```

Comment:
Same here: callers cannot call this concurrently with initialize_deferred_call_state.